dash/development/component_generator.py

```diff
@@ -22,7 +26,28 @@ class _CombinedFormatter(argparse.ArgumentDefaultsHelpFormatter,
 
 # pylint: disable=too-many-locals
 def generate_components(components_source, project_shortname,
-                        package_info_filename='package.json'):
+                        package_info_filename='package.json',
+                        generate_r_components=False,
+                        rprefix=None):
+    project_shortname = project_shortname.replace('-', '_').rstrip('/\\')
+
+    # import component library module
+    importlib.import_module(project_shortname)
```

Comment:
what does this do?

Reply:
Import the module into the namespace so it's available in `sys.module`.

Reply:
Yes, that I knew, but why are we doing it here?

Reply:
It is not imported before as we only know the module name with an argument and need to access the module to copy the `_js_dist`/`_css_dist` for dashR to use. The usage is actually in `generate_js_metadata` so it should be moved there for clarity.